{
    "expand": "operations,versionedRepresentations,editmeta,changelog,renderedFields",
    "fields": {
        "aggregateprogress": {
            "progress": 0,
            "total": 0
        },
        "aggregatetimeestimate": null,
        "aggregatetimeoriginalestimate": null,
        "aggregatetimespent": null,
        "assignee": null,
        "components": [],
        "created": "2007-11-15T08:08:09.000+0000",
        "creator": {
            "active": true,
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=dhruba&avatarId=30636",
                "24x24": "https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=dhruba&avatarId=30636",
                "32x32": "https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=dhruba&avatarId=30636",
                "48x48": "https://issues.apache.org/jira/secure/useravatar?ownerId=dhruba&avatarId=30636"
            },
            "displayName": "dhruba borthakur",
            "key": "dhruba",
            "name": "dhruba",
            "self": "https://issues.apache.org/jira/rest/api/2/user?username=dhruba",
            "timeZone": "America/Tijuana"
        },
        "customfield_10010": null,
        "customfield_12310191": null,
        "customfield_12310192": null,
        "customfield_12310220": "2007-11-28T16:08:38.637+0000",
        "customfield_12310222": "10002_*:*_2_*:*_126472859_*|*_1_*:*_2_*:*_1131639527_*|*_6_*:*_1_*:*_0_*|*_5_*:*_1_*:*_6141682627",
        "customfield_12310230": null,
        "customfield_12310250": null,
        "customfield_12310290": null,
        "customfield_12310291": null,
        "customfield_12310300": null,
        "customfield_12310310": "1.0",
        "customfield_12310320": null,
        "customfield_12310420": "19001",
        "customfield_12310920": "105797",
        "customfield_12310921": null,
        "customfield_12311020": null,
        "customfield_12311024": null,
        "customfield_12311120": null,
        "customfield_12311820": "0|i0igon:",
        "customfield_12312022": null,
        "customfield_12312023": null,
        "customfield_12312024": null,
        "customfield_12312026": null,
        "customfield_12312220": null,
        "customfield_12312320": null,
        "customfield_12312321": null,
        "customfield_12312322": null,
        "customfield_12312323": null,
        "customfield_12312324": null,
        "customfield_12312325": null,
        "customfield_12312326": null,
        "customfield_12312327": null,
        "customfield_12312328": null,
        "customfield_12312329": null,
        "customfield_12312330": null,
        "customfield_12312331": null,
        "customfield_12312332": null,
        "customfield_12312333": null,
        "customfield_12312334": null,
        "customfield_12312335": null,
        "customfield_12312336": null,
        "customfield_12312337": null,
        "customfield_12312338": null,
        "customfield_12312339": null,
        "customfield_12312340": null,
        "customfield_12312341": null,
        "customfield_12312520": null,
        "customfield_12312521": "Fri Nov 30 12:17:02 UTC 2007",
        "customfield_12312720": null,
        "customfield_12312823": null,
        "customfield_12312920": null,
        "customfield_12312921": null,
        "customfield_12312923": null,
        "customfield_12313422": "false",
        "customfield_12313520": null,
        "description": "I saw a case when the SecondaryNamenode encountered a OutOfMemory exception because the fsimage and edits file were too big to fit into the Secondary Namenode's memory. However, the process did not exit.\n\nException in thread \"org.apache.hadoop.dfs.SecondaryNameNode@73eb904d\" java.lang.OutOfMemoryError: GC overhead limit\nexceeded\n  at java.lang.AbstractStringBuilder.<init>(AbstractStringBuilder.java:45)\n  at java.lang.StringBuffer.<init>(StringBuffer.java:91)\n  at org.apache.hadoop.io.UTF8.toString(UTF8.java:129)\n  at org.apache.hadoop.dfs.FSImage.loadFSImage(FSImage.java:692)\n  at org.apache.hadoop.dfs.SecondaryNameNode.doMerge(SecondaryNameNode.java:304)\n  at org.apache.hadoop.dfs.SecondaryNameNode.doCheckpoint(SecondaryNameNode.java:273)\n  at org.apache.hadoop.dfs.SecondaryNameNode.run(SecondaryNameNode.java:190)\n  at java.lang.Thread.run(Thread.java:619)\n\t\t\n",
        "duedate": null,
        "environment": null,
        "fixVersions": [{
            "archived": false,
            "description": "",
            "id": "12312740",
            "name": "0.16.0",
            "releaseDate": "2008-02-07",
            "released": true,
            "self": "https://issues.apache.org/jira/rest/api/2/version/12312740"
        }],
        "issuelinks": [],
        "issuetype": {
            "avatarId": 21133,
            "description": "A problem which impairs or prevents the functions of the product.",
            "iconUrl": "https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21133&avatarType=issuetype",
            "id": "1",
            "name": "Bug",
            "self": "https://issues.apache.org/jira/rest/api/2/issuetype/1",
            "subtask": false
        },
        "labels": [],
        "lastViewed": null,
        "priority": {
            "iconUrl": "https://issues.apache.org/jira/images/icons/priorities/major.svg",
            "id": "3",
            "name": "Major",
            "self": "https://issues.apache.org/jira/rest/api/2/priority/3"
        },
        "progress": {
            "progress": 0,
            "total": 0
        },
        "project": {
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=12310240&avatarId=10095",
                "24x24": "https://issues.apache.org/jira/secure/projectavatar?size=small&pid=12310240&avatarId=10095",
                "32x32": "https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=12310240&avatarId=10095",
                "48x48": "https://issues.apache.org/jira/secure/projectavatar?pid=12310240&avatarId=10095"
            },
            "id": "12310240",
            "key": "HADOOP",
            "name": "Hadoop Common",
            "projectCategory": {
                "description": "Scalable Distributed Computing",
                "id": "10292",
                "name": "Hadoop",
                "self": "https://issues.apache.org/jira/rest/api/2/projectCategory/10292"
            },
            "self": "https://issues.apache.org/jira/rest/api/2/project/12310240"
        },
        "reporter": {
            "active": true,
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=dhruba&avatarId=30636",
                "24x24": "https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=dhruba&avatarId=30636",
                "32x32": "https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=dhruba&avatarId=30636",
                "48x48": "https://issues.apache.org/jira/secure/useravatar?ownerId=dhruba&avatarId=30636"
            },
            "displayName": "dhruba borthakur",
            "key": "dhruba",
            "name": "dhruba",
            "self": "https://issues.apache.org/jira/rest/api/2/user?username=dhruba",
            "timeZone": "America/Tijuana"
        },
        "resolution": {
            "description": "A fix for this issue is checked into the tree and tested.",
            "id": "1",
            "name": "Fixed",
            "self": "https://issues.apache.org/jira/rest/api/2/resolution/1"
        },
        "resolutiondate": "2007-11-29T21:36:41.000+0000",
        "status": {
            "description": "The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.",
            "iconUrl": "https://issues.apache.org/jira/images/icons/statuses/closed.png",
            "id": "6",
            "name": "Closed",
            "self": "https://issues.apache.org/jira/rest/api/2/status/6",
            "statusCategory": {
                "colorName": "green",
                "id": 3,
                "key": "done",
                "name": "Done",
                "self": "https://issues.apache.org/jira/rest/api/2/statuscategory/3"
            }
        },
        "subtasks": [],
        "summary": "SecondaryNamenode process should exit if it encounters Runtime exceptions",
        "timeestimate": null,
        "timeoriginalestimate": null,
        "timespent": null,
        "updated": "2011-09-27T11:44:50.000+0000",
        "versions": [],
        "votes": {
            "hasVoted": false,
            "self": "https://issues.apache.org/jira/rest/api/2/issue/HADOOP-2209/votes",
            "votes": 0
        },
        "watches": {
            "isWatching": false,
            "self": "https://issues.apache.org/jira/rest/api/2/issue/HADOOP-2209/watchers",
            "watchCount": 0
        },
        "workratio": -1
    },
    "id": "12382565",
    "key": "HADOOP-2209",
    "self": "https://issues.apache.org/jira/rest/api/2/issue/12382565"
}